op-node/rollup/derive/payload_attributes.go

```diff
@@ -224,32 +224,35 @@ func UserDeposits(receipts []*types.Receipt, depositContractAddr common.Address)
 	return out, errs
 }
 
-func BatchesFromEVMTransactions(config *rollup.Config, txLists []types.Transactions) ([]*BatchData, error) {
+func BatchesFromEVMTransactions(config *rollup.Config, txLists []types.Transactions) ([]*BatchData, []error) {
 	var out []*BatchData
+	var errs []error
 	l1Signer := config.L1Signer()
-	for _, txs := range txLists {
-		for _, tx := range txs {
+	for i, txs := range txLists {
+		for j, tx := range txs {
 			if to := tx.To(); to != nil && *to == config.BatchInboxAddress {
 				seqDataSubmitter, err := l1Signer.Sender(tx) // optimization: only derive sender if To is correct
 				if err != nil {
-					// TODO: log error
+					errs = append(errs, fmt.Errorf("invalid signature: tx list: %d, tx: %d, err: %w", i, j, err))
 					continue // bad signature, ignore
 				}
 				// some random L1 user might have sent a transaction to our batch inbox, ignore them
 				if seqDataSubmitter != config.BatchSenderAddress {
-					// TODO: log/record metric
+					errs = append(errs, fmt.Errorf("unauthorized batch submitter: tx list: %d, tx: %d", i, j))
 					continue // not an authorized batch submitter, ignore
 				}
 				batches, err := DecodeBatches(config, bytes.NewReader(tx.Data()))
 				if err != nil {
-					// TODO: log/record metric
+					errs = append(errs, fmt.Errorf("invalid batch: tx list: %d, tx: %d, err: %w", i, j, err))
 					continue
 				}
 				out = append(out, batches...)
+			} else {
+				errs = append(errs, fmt.Errorf("invalid to address: tx list: %d, tx: %d", i, j))
```

Comment:
This reads weird, maybe:

`invalid "to" address`

or

`invalid target`

Reply:
removed error